apps/namadillo/src/hooks/useValidateApiUrl.ts

```diff
@@ -0,0 +1,25 @@
+import { isUrlValid } from "@namada/utils";
+import { isApiAlive } from "atoms/settings";
+import { useMemo } from "react";
+
+export const useValidateApiUrl = (): ((url: string) => Promise<string>) => {
+  return useMemo(() => {
```

Comment:
Should we use useCallback here?

Reply:
yes! sorry, I was moving code around and forgot this